userspace/libsinsp/metrics_collector.cpp

```diff
@@ -485,7 +485,11 @@ void libs_metrics_collector::snapshot()
 	// METRICS_V2_STATE_COUNTERS related
 	uint64_t n_fds = 0;
 	uint64_t n_threads = 0;
-	std::shared_ptr<const sinsp_stats_v2> sinsp_stats_v2 = m_inspector->get_sinsp_stats_v2();
+	if (!m_sinsp_stats_v2)
+	{
+		// Sinsp stats disabled.
+		return;
```

Comment:
Below we also have other metrics categories that then could not be used without the sinsip stats?

Reply:
I think this early return is ok, it has the exact same behavior as before as far as i can tell; otherwise, can you point me to the lines (perhaps i am just missing them!)

Reply:
https://github.com/falcosecurity/libs/blob/570b86c1d72783b888ff4c3131159fbb1bc628c7/userspace/libsinsp/metrics_collector.cpp#L724-L785

Reply:
Fixed :)